Sources/Subprocess/Buffer.swift

```diff
@@ -150,12 +104,53 @@ extension AsyncBufferSequence.Buffer: Equatable, Hashable {
 
     public func hash(into hasher: inout Hasher) {
         self.data.withUnsafeBytes { ptr in
-            let bytes = UnsafeRawBufferPointer(
-                start: ptr,
-                count: self.data.count
-            )
-            hasher.combine(bytes: bytes)
+            hasher.combine(bytes: ptr)
         }
     }
     #endif
 }
+
+// MARK: - DispatchData.Block
+#if canImport(Darwin) || canImport(Glibc) || canImport(Android) || canImport(Musl)
+extension DispatchData {
+    /// Unfortunitely `DispatchData.Region` is not available on Linux, hence our own wrapper
+    internal struct Slice: @unchecked Sendable, RandomAccessCollection {
+        typealias Element = UInt8
+
+        internal let bytes: UnsafeBufferPointer<UInt8>
+
+        internal var startIndex: Int { self.bytes.startIndex }
+        internal var endIndex: Int { self.bytes.endIndex }
+
+        internal init(bytes: UnsafeBufferPointer<UInt8>) {
```

Comment:
Is this missing a startIndex argument? without that, 
```
internal var startIndex: Int { self.bytes.startIndex }
```
This will always return 0 since `UnsafeBufferPointer.startIndex` is always 0. 

And without that it would not match the semantics of the other Slice-type, where the index of the containing collection is preserved in the subsequence (like `ArraySlice` and `Array`) right?

Reply:
Bikeshed: As we talked previously, since this is not really representing a sub-sequence like a `Substring` or `ArraySlice`, can we not call it `Slice`? It sounds like we're really interested in ensuring that it's contiguous, so perhaps `ContiguousRegion` or something like that?